core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/cache/AiCacheStore.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkiverse.langchain4j.runtime.cache;
+
+import java.time.Instant;
+import java.util.List;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.message.AiMessage;
+
+/**
+ * Represents a store for the {@link AiCache} state.
+ * Allows for flexibility in terms of where and how cache is stored.
+ */
+public interface AiCacheStore {
+
+    public record CacheRecord(Embedding embedded, AiMessage response, Instant creation) {
```

Comment:
I think we should also store the embedding of the system message (if it existed), not just the user message.

Because if the system message is different, then even when the user message is very similar, we shouldn't use the cache for answering.

Reply:
Yes, I'm not worried about that in this "first version". I will add SystemMessage in the next commits.

Reply:
Fixed with this [commit](https://github.com/quarkiverse/quarkus-langchain4j/pull/659/commits/631e9043c13984edeb8f9b7b44e90ed6daf7a9a2#diff-2dd3bec40934ad6d175f6f14dad1af0e11c234cf5fec69739a89460d472ab55b).